strawberryfields/engine.py

```diff
@@ -168,11 +174,108 @@ def _run_program_locally(self, prog, **kwargs):
         Returns:
             list[Command]: commands that were applied to the backend
         """
+
+    def _run(self, program, *, compile=True, **kwargs):
+        """Execute the given programs by sending them to the backend.
+
+        For each :class:`Program` instance given as input, the following happens:
+
+        * The Program instance is compiled and optimized.
+        * The compiled program is executed on the backend, updating the backend state.
+        * The latest measurement results of each subsystem are stored
+          in the :class:`.RegRef` instances of the corresponding Program, as well as in self.samples.
+        * The compiled program is appended to self.run_progs.
+
+        Finally, the result of the computation is returned.
+
+        Args:
+            program (Program, Sequence[Program]): quantum programs to run
+            compile (bool): If True, compile the Program instances before sending them to the backend.
```

Comment:
This way the user has to specifically overwrite the default option. Since `compile` actually means `compile_and_optimize`, I can see a use-case for this (e.g., comparing two programs which are valid, but one is unoptimized)

Reply:
How about if I replace the `compile` parameter with a kwargs-like dictionary `compile_options` which is passed on to `Programs.compile()`?

Reply:
I like that solution :+1:

Reply:
Sounds good @smite